File: CoreJava/Shoppingcart-app/src/com/techlabs/shoppingcart/LineItem.java
package com.techlabs.shoppingcart;

public class LineItem {
	private int id;
	private int quantity;
	private Product products;

	public LineItem(int id, int quantity, Product product) {
		super();
		this.id = id > 0 ? id : 0;
		this.quantity = quantity > 0 ? quantity : 0;
		this.products = product;
	}

	public int getId() {
		return id;
	}

	public int getQuantity() {
		return quantity;
	}

	public Product getProduct() {
		return products;
	}

	public double costofLineItem() {
		return products.calcDiscountCost() * getQuantity();
	}

}

File: DotNET/C#/ShoppingCartApp/ShoppingCartApp/LineItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ShoppingCartApp
{
    class LineItem
    {
        private int _id;
        private int _qunatity;
        private Product _products;


        public LineItem(int id, int quantity, Product product)
        {
            _id = id;
            _qunatity = quantity;
            _products = product;
        }

        public int Id
        {
            get
            {
                return _id;
            }
        }

        public int Quantity
        {
            get
            {
                return _qunatity;
            }
        }

        public Product Product
        {
            get
            {
                return _products;
            }
        }

        public double costofLineItem()
        {
            return _products.calcDiscountCost() * Quantity;
        }

    }

}
